apm-agent-core/src/main/java/co/elastic/apm/agent/impl/context/TransactionContext.java

```diff
@@ -55,17 +59,34 @@ public void copyFrom(TransactionContext other) {
         user.copyFrom(other.user);
     }
 
-    /**
-     * An arbitrary mapping of additional metadata to store with the event.
-     */
-    public Map<String, Object> getCustom() {
-        return custom;
+    public Object getCustom(String key) {
+        return custom.get(key);
     }
 
     public Response getResponse() {
         return response;
     }
 
+    public void addCustom(String key, String value) {
+        custom.put(key, value);
```

Comment:
No size limitation at all on those?

Reply:
the size limitation is enforced centrally on serialization

Reply:
Ahh. Don't we want to allow a bigger size for these? Probably a common use for non-indexed text is big texts that users still want to keep, like HTTP bodies.

Reply:
Oh, good point. We'll want to use the 10k char limit on those values!